fetch_depth_layer/src/depth_layer.cpp

```diff
@@ -395,4 +398,16 @@ void FetchDepthLayer::depthImageCallback(
   }
 }
 
+void FetchDepthLayer::activate()
+{
+  onInitialize();
```

Comment:
onInitialize will also re-collect all of the params, make new marking and clearing buffers, and make new publishers.  Is this desired?